### PR TITLE
docs: add components sidebar

### PR DESCRIPTION
**Pull request by MadCcc**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Sidebar      |
| 🇨🇳 Chinese |        -   |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: .dumi/theme/slots/Sidebar/index.tsx [hunk 1]
```diff
@@ -8,11 +12,59 @@ const Sidebar: FC = () => {
...
+      } else {
+        items.push(
+          ...group.children?.map(item => ({
+            label: <Link to={item.link}>{item.title}</Link>,
+            key: item.link.replace(/(-cn$)/g, ''),
+          })),
+        );
+      }
+    }
+    return items;
```
Comment (afc163) [inline .dumi/theme/slots/Sidebar/index.tsx:41]: 用个 map？

File: .dumi/pages/components/overview.tsx [hunk 2]
```diff
@@ -1,3 +1,8 @@
+/**
+ * title: Components Overview
+ * description: `antd` provides plenty of UI components to enrich your web applications, and we will improve components experience consistently. We also recommend some great [Third-Party Libraries](/docs/react/recommendation) additionally.
+ */
```
Comment (afc163) [inline .dumi/pages/components/overview.tsx:4]: 不能放 md 里么？

Comment (afc163) [reply .dumi/pages/components/overview.tsx:4]: 感觉应该反过来，md 里放 react 组件。

https://github.com/ant-design/ant-design/pull/37916/files#diff-c888c5e57336735e015b7347d89f3d9b1c3c3676c54576a643c3cab7f2aa2ef8

Comment (MadCcc) [reply .dumi/pages/components/overview.tsx:4]: 这个页面被 dumi 约定了，md 可以不用了

Comment (MadCcc) [reply .dumi/pages/components/overview.tsx]: 改成 md 引入了

**Pull request merged**